packages/ui/lib/utils.ts

```diff
@@ -6,3 +6,9 @@ export const minifyAddress = (address: string) => {
     ? `${address.slice(0, 5)}...${address.slice(address.length - 5)}`
     : address
 }
+
+export const isAddressOrEns = (address = '') => {
+  return (
+    address?.toLowerCase()?.includes('.eth') || ethers.utils.isAddress(address)
```

Comment:
Kali, ENS names don't necessarily end up with `.eth`.... So I don't think that's right. If you want to check if something is a valid ENS you need to check that it resolves correctly.

Reply:
making this an async function will cause a side-effect in the component, the idea here is to have a result without calling `resolveName` and waiting for the result.